Source/UIViewController+SemiModalViewController.swift

```diff
@@ -59,14 +59,15 @@ extension UIViewController {
     public func presentSemiView(_ view: UIView, options: [SemiModalOption: Any]? = nil, completion: (() -> Void)? = nil) {
         registerOptions(options)
         let targetView = parentTargetView()
-        
+        let targetParentVC = parentTargetViewController()
+
         if targetView.subviews.contains(view) {
             return
         }
         
         objc_setAssociatedObject(view, &semiModalPresentingViewController, self, .OBJC_ASSOCIATION_RETAIN_NONATOMIC)
         
-        NotificationCenter.default.addObserver(self,
+        NotificationCenter.default.addObserver(targetParentVC,
```

Comment:
vc|childvc
-----|-------
when presentSemiView <br> `addObserver self` <br> (self is vc)|  in childVC  <br>`dismissSemiModalViewWithCompletion` <br> `removeObserver on self` <br> (self is childvc)

So `NotificationCenter` add and remove always on TargetVC , it will can be removed.

if do not do this, `NotificationCenter`  will maybe be added, but do not be removed